### PR TITLE
1269 funnel trends 2

### PR DESCRIPTION
**Pull request by timgl**
## Changes

Adds toggle between funnel viz and funnel trends. Behind the `funnel-trends-1269` feature flag

todo
- [x] hide interval in funnel mode
- [x] feature flag
- [ ] people on click (maybe another pr as this is already big)
- [x] fix dashboard

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: frontend/src/scenes/funnels/funnelVizLogic.js [hunk 1]
```diff
@@ -8,12 +8,14 @@ export const funnelVizLogic = kea({
     loaders: ({ props }) => ({
         results: {
             __default: [],
-            loadResults: async (refresh = false) => {
+            // TODO BEFORE COMMITTING SET REFRESH TO FALSE
+            loadResults: async (refresh = true) => {
```
Comment (EDsCODE) [inline frontend/src/scenes/funnels/funnelVizLogic.js:12]: TODO here it seems

File: frontend/src/scenes/insights/Insights.js [hunk 2]
```diff
@@ -224,10 +239,10 @@ function _Insights() {
                                 <Card
                                     title={
                                         <div className="float-right">
-                                            {showIntervalFilter[activeView] && (
+                                            {showIntervalFilter(allFilters) && (
```
Comment (EDsCODE) [inline frontend/src/scenes/insights/Insights.js:242]: Could be irrelevant as I can't reproduce it, but when i first loaded this PR, it seemed to lag before the filters were shown which could have to do with the filter logic depending on `allfilters` now

**Review by EDsCODE — COMMENTED** [review]
Initial QA to catch some bugs and see how it worked and testing on a local PG instance doesn't load a graph. Seems to timeout.

WIll follow with in depth backend review

**Comment by timgl** [comment]
@EDsCODE Can't reproduce your issue with loading the graph, how are you getting into it?

File: frontend/src/scenes/insights/Insights.js [hunk 3]
```diff
@@ -57,27 +58,41 @@ const displayMap = {
     [`${ACTIONS_LINE_GRAPH_CUMULATIVE}`]: CUMULATIVE_CHART_LABEL,
     [`${ACTIONS_TABLE}`]: TABLE_LABEL,
     [`${ACTIONS_PIE_CHART}`]: PIE_CHART_LABEL,
-    [`${ACTIONS_BAR_CHART}`]: BAR_CHART_LABEL,
+    [`${FUNNEL_VIZ}`]: FUNNEL_LABEL,
```
Comment (EDsCODE) [inline frontend/src/scenes/insights/Insights.js:61]: don't think the bar chart label was supposed to get tossed

**Review by EDsCODE — COMMENTED** [review]
QA'd on both clickhouse and pg. Looks good. Few remaining comments

- can't seem to get a single step funnel trend to make sense. If i'm understanding funnel trends properly, I think it should be 100% on the days there are events, but it's always 0. 
- also it would be helpful to bring through the absolute numbers so the percentages have context, otherwise i'm not sure what numbers to be thinking about on the graph

**Review by Twixes — CHANGES_REQUESTED** [review]
How trends can be enabled here is rather unintuitive, as "Funnel" for the original viz is redundant (the user already knows they're working on a Funnel type insight) and "Line graph" for trends is much too vague. The selector is also confusing because of the inclusion of all the other, completely not applicable options. The choice here should preferably be only between "Steps" and "Trends".
Confusing options

The UX of analysis itself is weird due to multiple funnel steps being allowed. Unknown behavior with more than two steps also causes this to be largely incompatible with existing funnels (hard to say between which steps is the trends analysis applied, and you cannot select).
To alleviate this, somewhere here could be options "From step" and "To step" (vide e.g. Mixpanel).
Display settings

As a general point regarding funnels, we should not be allowing calculation of single-step funnels, as they can only result in useless and confusing behavior (Re: @EDsCODE notes).

**Comment by yakkomajuri** [comment]
Flyby comment: If I click 'Linear' and then click on a saved funnel, it loads the funnel normally rather than the trend. I need to switch away and back for it to work. I guess we're expecting a funnel to load before the user flips 'Linear'

**Comment by timgl** [comment]
Thanks for feedback all (and sorry for delay)

- Have simplified ChartFilter to only show Trends and Steps as options
- Now showing an error if you only have one step with trends
image

- Will do the percentages in a different PR. I did add a little explainer of what you're looking at:
image

@yakkomajuri I think this makes sense? If you've saved a funnel as a Steps funnel you'll want to see it that way the next time?

**Comment by yakkomajuri** [comment]
@timgl come to think about it that does make sense - gave the deployed env a new spin though and have been getting 500s when switching around different settings (range, steps/trends, period)

[See this link](https://posthog-1269-funnel-tre-uml3u6.herokuapp.com/api/insight/funnel/?insight=FUNNELS&properties=%5B%5D&display=FunnelViz&interval=week&actions=%5B%7B%22id%22%3A16%2C%22math%22%3Anull%2C%22name%22%3A%22Installed%20App%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A0%2C%22properties%22%3A%5B%5D%2C%22math_property%22%3Anull%7D%2C%7B%22id%22%3A15%2C%22math%22%3Anull%2C%22name%22%3A%22Rated%20App%22%2C%22type%22%3A%22actions%22%2C%22order%22%3A1%2C%22properties%22%3A%5B%7B%22key%22%3A%22plan%22%2C%22type%22%3A%22event%22%2C%22value%22%3A%22premium%22%2C%22operator%22%3Anull%7D%5D%2C%22math_property%22%3Anull%7D%5D&date_from=all)

**Comment by timgl** [comment]
@yakkomajuri actually looks like those are just redis errors, for some reason it's reaching the max number of clients https://sentry.io/organizations/hiberly-nm/issues/2200733211/?project=5624479&query=is%3Aunresolved#

**Comment by yakkomajuri** [comment]
Ah yes our deployment envs use a hobby dev redis with a 20 connection limit. We reach this limit super fast - had this problem on the playground at first.

**Pull request merged**